### PR TITLE
Feature/itbl 3347 universal link redirects

### PR DESCRIPTION
**Pull request by davidtruong**
Added a URLSession delegate which handles the dataTaskWithURL task. Stops the request after a response and doesn't make the redirect call. Returns the redirected url to the callback of the getAndTrackRequest call.

### CODE REVIEW

**Comment by davidtruong** [comment]
@dontgitit Any thoughts on if I should leave getAndTrackDeeplink as a instance function vs making it static? It can be an instance function because getAndTrackRedirect gets called in continueUserActivity after IterableApi is created in application willFinishLaunchingWithOptions.

Originally I made it an instance so that it could share the same session. However it turns out that the request should be on a different session so it doesn't break our other api calls if they happen to make a redirect.

File: Iterable-iOS-SDK/IterableDeeplink.m [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+@end
+
+@implementation IterableDeeplink {
+}
+
+// the URL session we're going to be using
+static NSURLSession *redirectUrlSession = nil;
+NSString *deepLinkLocation = nil;
+
+// documented in IterableAPI.h
```
Comment (dontgitit) [inline Iterable-iOS-SDK/IterableDeeplink.m]: IterableDeeplink.h

**Review by dontgitit — APPROVED** [review]
lftm assuming it works

**Comment by dontgitit** [comment]
@davidtruong whatever makes the most sense/is easiest to use for people implementing it

**Comment by davidtruong** [comment]
Refactored the class to be called IterableDeeplinkManager and use a singleton.
Changed getAndTrackDeeplink back to be static so that is backwards compatible with applications that already have the sdk integrated.

**Pull request merged**